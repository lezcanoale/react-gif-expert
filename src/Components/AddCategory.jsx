import { useState } from "react";
import PropTypes from 'prop-types';
export const AddCategory = ({onNewCategory}) => {

    const [	inputValue, setinputValue] = useState('');
    
    const onInputChange=({target})=>{
      setinputValue(target.value);
    }

    const onSubmit=(event)=>{
      console.log('Submit del form');
      event.preventDefault();
      if(inputValue.trim().length<=1) return;
      // setCategories(categories=>[...categories,inputValue]);
      onNewCategory(inputValue.trim());
      setinputValue('');
    }

      return (
        <form onSubmit={onSubmit} role="form">
          <input 
            type="text" 
            placeholder="Buscar Gifs" 
            value={inputValue}
            onChange={onInputChange}>
          </input>
        </form>
      )
}
AddCategory.propTypes={
  onNewCategory:PropTypes.func.isRequired,
}


