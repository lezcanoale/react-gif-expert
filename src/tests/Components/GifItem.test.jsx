import { GifItem } from "../../Components/GifItem";
import {render,screen} from '@testing-library/react';
describe('Prueba GifItem', () => {
  const title='StrangerThings';
  const url='https://hoypyspace.sfo2.cdn.digitaloceanspaces.com/imagenes/stranger-things-temporada-4-netflix.jpg';
  test('El componente debe coincidir con el snapshot', () => {
    const {container}=render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });
  test('debe mostrar la imagen con el URL y el ALT indicado', () => {
    render(<GifItem title={title} url={url}/>);
    // screen.debug();
    // expect(screen.getAllByRole('img').src);
    // console.log(screen.getByRole('img').alt);
    // expect(screen.getByRole('img').alt).toBe(title);
    // expect(screen.getByRole('img').src).toBe(url);
    const{ src,alt}=screen.getByRole('img');
    expect(alt).toBe(alt);
    expect(src).toBe(url);
  });
  test('debe de mostrar el titulo en el componente', () => {
    render(<GifItem title={title} url={url}/>);
    // expect(screeN.getByRole('p'))
    // console.log(screen.getByText(title));
    expect(screen.getByText(title)).toBeTruthy();
  });
  
  
})
