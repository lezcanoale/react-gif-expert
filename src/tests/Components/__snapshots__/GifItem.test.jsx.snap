// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba GifItem El componente debe coincidir con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      src="https://hoypyspace.sfo2.cdn.digitaloceanspaces.com/imagenes/stranger-things-temporada-4-netflix.jpg"
    />
    <p>
      StrangerThings
    </p>
  </div>
</div>
`;
